types/jest/index.d.ts

```diff
@@ -28,1557 +28,22 @@
-         */
-        stub(): Spy;
-    }
-
-    interface Calls {
-        /**
-         * By chaining the spy with calls.any(),
-         * will return false if the spy has not been called at all,
-         * and then true once at least one call happens.
-         */
-        any(): boolean;
-        /**
-         * By chaining the spy with calls.count(),
-         * will return the number of times the spy was called
-         */
-        count(): number;
-        /**
-         * By chaining the spy with calls.argsFor(),
-         * will return the arguments passed to call number index
-         */
-        argsFor(index: number): any[];
-        /**
-         * By chaining the spy with calls.allArgs(),
-         * will return the arguments to all calls
-         */
-        allArgs(): any[];
-        /**
-         * By chaining the spy with calls.all(), will return the
-         * context (the this) and arguments passed all calls
-         */
-        all(): CallInfo[];
-        /**
-         * By chaining the spy with calls.mostRecent(), will return the
-         * context (the this) and arguments for the most recent call
-         */
-        mostRecent(): CallInfo;
-        /**
-         * By chaining the spy with calls.first(), will return the
-         * context (the this) and arguments for the first call
-         */
-        first(): CallInfo;
-        /**
-         * By chaining the spy with calls.reset(), will clears all tracking for a spy
-         */
-        reset(): void;
-    }
-
-    interface CallInfo {
-        /**
-         * The context (the this) for the call
-         */
-        object: any;
-        /**
-         * All arguments passed to the call
-         */
-        args: any[];
-        /**
-         * The return value of the call
-         */
-        returnValue: any;
-    }
-
-    interface CustomMatcherFactories {
-        [index: string]: CustomMatcherFactory;
-    }
-
-    type CustomMatcherFactory = (util: MatchersUtil, customEqualityTesters: CustomEqualityTester[]) => CustomMatcher;
-
-    interface MatchersUtil {
-        equals(a: any, b: any, customTesters?: CustomEqualityTester[]): boolean;
-        // tslint:disable-next-line: no-unnecessary-generics
-        contains<T>(haystack: ArrayLike<T> | string, needle: any, customTesters?: CustomEqualityTester[]): boolean;
-        buildFailureMessage(matcherName: string, isNot: boolean, actual: any, ...expected: any[]): string;
-    }
-
-    type CustomEqualityTester = (first: any, second: any) => boolean;
-
-    interface CustomMatcher {
-        compare<T>(actual: T, expected: T, ...args: any[]): CustomMatcherResult;
-        compare(actual: any, ...expected: any[]): CustomMatcherResult;
-    }
-
-    interface CustomMatcherResult {
-        pass: boolean;
-        message: string | (() => string);
-    }
-
-    interface ArrayLike<T> {
-        length: number;
-        [n: number]: T;
-    }
-}
+// Minimum TypeScript Version: 4.5
```

Comment:
Should be 4.3 (only jsdom types require 4.5).

Reply:
All should be fine with 4.2 as well. If it helps to resolve errors in depends.

Type tests are running on 4.3 in Jest repo. The limitation comes from current setup, but it might be possible to rework it. I just didn't find time yet (;

Reply:
It failed on `jest-mock` for lower versions.

Reply:
Interesting! Sounds like a bug - as mentioned we use TS 4.3 in our own type tests